app/graphql/types/query_type.rb

```diff
@@ -160,6 +178,23 @@ def dynamic_field(id:)
       dynamic_field
     end
 
+    def enabled_dynamic_fields(project:, digital_object_type:)
+      project = Project.find_by!(project.to_h)
+      ability.authorize!(:read, project)
+      join_query = "SELECT dynamic_fields.id AS df_id, enabled_dynamic_fields.* FROM dynamic_fields"\
+                   " LEFT OUTER JOIN enabled_dynamic_fields ON enabled_dynamic_fields.dynamic_field_id = dynamic_fields.id"\
+                   " AND enabled_dynamic_fields.project_id = #{project.id} AND enabled_dynamic_fields.digital_object_type = '#{digital_object_type}'"
+      field_sets = project.field_sets.where(enabled_dynamic_fields: { digital_object_type: digital_object_type }).includes(:enabled_dynamic_fields)
+      DynamicField.connection.select_all(join_query).map do |result|
```

Comment:
Good use of raw SQL query.  :+1:

Do you think it would be worth moving the SQL query lines and `DynamicField.connection.select_all(join_query)` to a clearly-named static method on the DynamicField model (that would take two arguments -- project id and digital object type)?  It would compress this method a bit and would improve skim-ability.

Reply:
I think something like that is a good idea, but I'm not sure what class to put it on or what to call it - when I start thinking about the naming is awkward. `EnabledDynamicField.or_not_for_project` or `DynamicField.with_enabling_if_enabled`

Reply:
opened HYACINTH-646 to track this refactor.

Reply:
Thanks!